src/theory/quantifiers/sygus/ce_guided_conjecture.cpp

```diff
@@ -196,20 +205,25 @@ bool CegConjecture::isSingleInvocation() const {
 bool CegConjecture::needsCheck( std::vector< Node >& lem ) {
   if( isSingleInvocation() && !d_ceg_si->needsCheck() ){
     return false;
-  }else{
-    bool value;
-    Assert( !getGuard().isNull() );
-    // non or fully single invocation : look at guard only
-    if( d_qe->getValuation().hasSatValue( getGuard(), value ) ) {
-      if( !value ){
-        Trace("cegqi-engine-debug") << "Conjecture is infeasible." << std::endl;
-        return false;
-      }
-    }else{
-      Assert( false );
+  }
+  bool value;
+  Assert(!d_feasible_guard.isNull());
```

Comment:
Why you used `AlwaysAssert` after creating but now you use `Assert`?

Reply:
needsCheck is called frequently, this is for performance.

Reply:
ok